lib/cc/engine/markdownlint.rb

```diff
@@ -16,26 +17,24 @@ def initialize(root, engine_config, io)
 
       def run
         return if include_paths.strip.length == 0
-        run_mdl
-      end
-
-      private
-
-      attr_reader :root, :engine_config, :io, :contents
 
-      def run_mdl
         pid, _, out, err = POSIX::Spawn.popen4("mdl --no-warnings #{include_paths}")
         out.each_line do |line|
           io.print JSON.dump(issue(line))
           io.print "\0"
         end
       ensure
-        STDERR.print err.read
-        [out, err].each(&:close)
-
-        Process::waitpid(pid)
+        if pid
+          STDERR.print err.read
+          [out, err].each(&:close)
+          Process::waitpid(pid)
+        end
```

Comment:
Would you be willing to split out (what looks like) the behavior-neutral refactors into a pre-requisite PR?